.github/workflows/changelog-checker.yml

```diff
@@ -0,0 +1,36 @@
+# This workflow checks that there is either a 'pr/no-changelog' label applied to a PR
+# or there is a .changelog/<pr number>.txt file associated with a PR for a changelog entry
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled]
+    # Runs on PRs to master and all release branches
+    branches:
+      - master
+      - release/*
+
+jobs:
+  # checks that a .changelog entry is present for a PR
+  changelog-check:
+    # If there  a `pr/no-changelog` label we ignore this check
+    if: "!contains(github.event.pull_request.labels.*.name, 'pr/no-changelog')"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 # by default the checkout action doesn't checkout all branches
```

Comment:
Can we omit this now that it uses `on: pull_request` instead of target?

Reply:
I can test but I don't think so, we still need to checkout all branches to compare the HEAD to the base ref (which wouldn't be checked out if we just checkout HEAD).